lib/tapioca/generator.rb

```diff
@@ -534,12 +536,16 @@ def compile_gem_rbi(gem)
       say("Compiling #{gem_name}, this may take a few seconds... ")
 
       strictness = config.typed_overrides[gem.name] || "true"
-      rbi_body_content = compiler.compile(gem)
+      rbi = compiler.compile(gem)
+      rbi.comments << RBI::Comment.new("typed: #{strictness}") unless rbi.empty?
+      rbi.nest_singleton_methods!
+      rbi = Tapioca::RBIMerger.merge_with_exported_rbi!(rbi, gem, Thor::Shell::Color.new)
```

Comment:
Why the `!` suffix? None of the argument will be modified right?

Reply:
I'm not sure to understand why the `merge_with_exported_rbi` is extracted in a separate class here. The separation seems very artificial. For proof I'd cite the strong coupling with `Shell::Color.new`.

As I see it it would make more sense to keep it here, so it would be moved to the `Gem` generator once #504 is merged? 

If we do extract it as a separate service, it should also be used when we merge the gem's RBIs from the `rbi/` dir.

Reply:
> Why the ! suffix? None of the argument will be modified right?

Oversight on my part. At some point it was calling a bang method but indeed, not anymore. I can definitely just move it back to the other file though 👍